core/server/models/base/index.js

```diff
@@ -717,19 +712,19 @@ ghostBookshelf.Model = ghostBookshelf.Model.extend({
         }
 
         return itemCollection.fetchPage(options).then(function formatResponse(response) {
-            var data = {},
-                models;
+            const data = {};
+            data.meta = {pagination: response.pagination};
 
-            options.columns = requestedColumns;
-            models = response.collection.toJSON(options);
+            // Filtering attributes here so they are not leaked to api layer
+            // where models are serialized to json and do not do more filtering
+            data[tableName] = response.collection.models.map((model) => {
+                if (requestedColumns) {
+                    model.attributes = _.pick(model.attributes, requestedColumns);
```

Comment:
By using `requestedColumns` here, we're going to omit any columns from `defaultColumnsToFetch` that weren't specified. Is that intended? If so, it might be worth not fetching them in the first place?

Reply:
See https://github.com/TryGhost/Ghost/blob/master/core/server/models/post.js#L390

Reply:
Is this an intended change then? (not your PR i know :grin: )

Reply:
@allouis  This is an intended change. Before filtering was done in `toJSON` method with options being passed in. The filtering is meant to stay in this layer, but without calling `toJSON` there is no other good way to do it.

Reply:
:+1: